### PR TITLE
Encode path parameters in rpc calls

### LINKED ISSUE #20: Path parameters aren't encoded in rpc calls
Here are two examples with different codepaths where this causes a bug.

1) Reading a dataset branch:
```
import palantir.datasets as ptd
dataset = ptd.dataset('ri.foundry.main.dataset.1', branch='test/branch').read_pandas()
```
The obvious workaround doesn't work (replacing the branch name above with `test%2Fbranch`) as the branch name is also used in a SQL query.

2) Creating a dataset with a branch:
```
import palantir.datasets as ptd
new_dataset = ptd.dataset('/Users/ktekelioglu/new_dataset3', branch='test/branch', create=True)
```

### PR DESCRIPTION
**Pull request by ktekelioglu**
Fixes https://github.com/palantir/palantir-python-sdk/issues/20

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/palantir-python-sdk](https://github.com/palantir/palantir-python-sdk), @impossiblity! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: palantir/core/util.py [hunk 1]
```diff
@@ -59,3 +60,10 @@ def process_page(page):
                 yield value
 
     return process_page(page_supplier(page_token))
+
+
+def format_path_with_params(path, path_params):
```
Comment (ahggns) [inline palantir/core/util.py]: lets place this in a new util file in the rpc directory

File: palantir/datasets/rpc/catalog.py [hunk 2]
```diff
@@ -43,7 +44,7 @@ def create_dataset(
         _json: Any = ConjureEncoder().default(request)
 
         _path = "/catalog/datasets"
-        _path = _path.format(**_path_params)
+        _path = format_path_with_params(_path, _path_params)
```
Comment (ahggns) [inline palantir/datasets/rpc/catalog.py:47]: given this is codegen surely this must also be an issue in [conjure-python](https://github.com/palantir/conjure-python). Can you raise an issue (or better, port this fix) over there?

Comment (ktekelioglu) [reply palantir/datasets/rpc/catalog.py:47]: Oh, didn't know this was codegen. I'd also advocate for a header to be added on top of codegen files indicating as such.

**Comment by ktekelioglu** [comment]
@ahggns I'm happy to close this for now & iterate over at conjure-python. We can re-generate the rpc files once we have a fix there.

**Review by ahggns — APPROVED** [review]

**Comment by ktekelioglu** [comment]
@ahggns you also need to add a changelog, I do not have the perms to do so.

**Pull request merged**